atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/paging/PagingIterable.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra.paging;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.cassandra.thrift.ColumnOrSuperColumn;
+import org.apache.cassandra.thrift.KeyRange;
+import org.apache.cassandra.thrift.KeySlice;
+
+import com.google.common.base.Supplier;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RangeRequests;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.cassandra.ResultsExtractor;
+import com.palantir.util.paging.AbstractPagingIterable;
+import com.palantir.util.paging.SimpleTokenBackedResultsPage;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+
+public class PagingIterable<T, U>
+        extends AbstractPagingIterable<RowResult<U>, TokenBackedBasicResultsPage<RowResult<U>, byte[]>> {
+    private final ColumnGetter columnGetter;
+    private final RangeRequest rangeRequest;
+    private final Supplier<ResultsExtractor<T, U>> resultsExtractor;
+    private final long timestamp;
+
+    private final int batchHint;
+    private final ColumnSelection selection;
+    private final RowGetter rowGetter;
+
+    public PagingIterable(
+            RowGetter rowGetter,
+            ColumnGetter columnGetter,
+            RangeRequest rangeRequest,
+            Supplier<ResultsExtractor<T, U>> resultsExtractor,
+            long timestamp) {
+        this.rowGetter = rowGetter;
+        this.columnGetter = columnGetter;
+        this.rangeRequest = rangeRequest;
+        this.resultsExtractor = resultsExtractor;
+        this.timestamp = timestamp;
+
+        batchHint = rangeRequest.getBatchHint() == null ? 100 : rangeRequest.getBatchHint();
+        selection = rangeRequest.getColumnNames().isEmpty() ? ColumnSelection.all()
+                : ColumnSelection.create(rangeRequest.getColumnNames());
+    }
+
+    @Override
+    protected TokenBackedBasicResultsPage<RowResult<U>, byte[]> getFirstPage() throws Exception {
```

Comment:
Shouldn't we change `PagingIterable` (and `AbstractPagingIterable`) so that these methods are `public`? This seems super weird.